### PR TITLE
lib/vector: Replaced strcpy with G_strlcpy

### PR DESCRIPTION
**Pull request by ShubhamDesai**
This pull request addresses the potential buffer overflow issues by replacing all instances of strcpy with G_strlcpy.

### CODE REVIEW

**Comment by echoix** [comment]
I just thought about this by seeing the new header import:

Is there some circular dependencies introduced by needing the gis header for using G_strlcpy (not the function directly, but to display the error message)? What is the impact on additional code to link at compile time, and the extra libraries depended on?
It's more a general question for @nilason or others...

**Comment by wenzeslaus** [comment]
In this case, gis.h is already included by vector.h which is already included in the modified file, so I would say no change in dependencies here. In general, all these functions are in the general (aka gis) group, so you get one or all. I think it's the same for included and compile dependencies (linking).

File: lib/vector/diglib/frmt.c [hunk 1]
```diff
@@ -46,7 +48,10 @@ int dig_read_frmt_ascii(FILE *dascii, struct Format_info *finfo)
             return -1;
         }
 
-        strcpy(buf1, buff);
+        len = G_strlcpy(buf1, buff, sizeof(buf1));
+        if (len >= sizeof(buf1)) {
+            G_fatal_error(_("Name <%s> is too long"), buff);
+        }
```
Comment (wenzeslaus) [inline lib/vector/diglib/frmt.c:54]: This function's doc says it returns -1 on error, so it should not call fatal.

Couple lines above it gives warning on error and returns -1.

File: lib/vector/diglib/frmt.c [hunk 2]
```diff
@@ -46,7 +48,10 @@ int dig_read_frmt_ascii(FILE *dascii, struct Format_info *finfo)
             return -1;
         }
 
-        strcpy(buf1, buff);
+        len = G_strlcpy(buf1, buff, sizeof(buf1));
+        if (len >= sizeof(buf1)) {
+            G_fatal_error(_("Name <%s> is too long"), buff);
```
Comment (wenzeslaus) [inline lib/vector/diglib/frmt.c:53]: It is a _line_ which is too long, not a _name_.

Compare _G_getl2_ call on line 41/43 and the code in r.patch which uses "Name ...".

**Review by wenzeslaus — CHANGES_REQUESTED** [review]

File: lib/vector/diglib/frmt.c (hunk 1 above)
Comment (ShubhamDesai) [reply lib/vector/diglib/frmt.c]: done changes

File: lib/vector/diglib/frmt.c (hunk 2 above)
Comment (ShubhamDesai) [reply lib/vector/diglib/frmt.c]: done

**Review by wenzeslaus — APPROVED** [review]
This looks good to me. Thanks!

**Comment by nilason** [comment]
> I just thought about this by seeing the new header import:
> 
> Is there some circular dependencies introduced by needing the gis header for using G_strlcpy (not the function directly, but to display the error message)? What is the impact on additional code to link at compile time, and the extra libraries depended on? Is there an impact on compile time (if having cycles in our dependency graph)? It's more a general question for @nilason or others...

`diglib` (part of lib/vector) already has libgis as a dependency and uses libgis extensively, so no change in that respect. The reason this worked without explicitly including "grass/gis.h" is, as Vashek correctly mentioned, it is included by "grass/vector.h". I'd say it is generally advisable to include explicitly what is being used in a file, so adding the `#include <grass/gis.h>` is good, adds clarity.
(In every header file (at least in GRASS library) there is an "include guard" which makes sure a header file is only read once by the compiler.)

**Pull request merged**